platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/WorkflowStarter.java

```diff
@@ -0,0 +1,55 @@
+package com.softicar.platform.workflow.module.workflow;
+
+import com.softicar.platform.db.core.transaction.DbTransaction;
+import com.softicar.platform.workflow.module.workflow.item.AGWorkflowItem;
+import com.softicar.platform.workflow.module.workflow.item.IWorkflowableObject;
+import com.softicar.platform.workflow.module.workflow.task.WorkflowTaskManager;
+import java.util.Objects;
+
+public class WorkflowStarter {
+
+	private final AGWorkflow workflow;
+
+	public WorkflowStarter(AGWorkflow workflow) {
+
+		this.workflow = Objects.requireNonNull(workflow);
+	}
+
+	/**
+	 * Starts the workflow for the given {@link IWorkflowableObject}.
+	 * <p>
+	 * Executes auto transitions and generates tasks if appropriate.
+	 *
+	 * @param object
+	 *            the {@link IWorkflowableObject} (never <i>null</i>)
+	 */
+	public void startWorkflow(IWorkflowableObject<?> object) {
+
+		Objects.requireNonNull(object);
+
+		try (var transaction = new DbTransaction()) {
+			if (object.getWorkflowItem() == null) {
+				AGWorkflowItem item = new AGWorkflowItem() //
+					.setWorkflow(workflow)
+					.setWorkflowNode(workflow.getCurrentVersion().getRootNode())
+					.save();
+
+				object.setWorkflowItem(item);
+				object.save();
+			} else {
+				object//
```

Comment:
This essentially restarts the workflow if theres already a WorkflowItem present, doesnt it?
If so, this should definitely be described in the javadoc

Reply:
We should handle that in a separate issue / PR since the outlined behavior was unchanged.